qiskit_optimization/applications/sk_model.py

```diff
@@ -0,0 +1,108 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the Sherrington Kirkpatrick (SK) model."""
+
+from typing import Union, List, Optional
+
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from .graph_optimization_application import OptimizationApplication
+from ..algorithms import OptimizationResult
+from ..problems.quadratic_program import QuadraticProgram
+from ..translators import from_docplex_mp
+
+
+class SKModel(OptimizationApplication):
+    r"""Optimization application of the "Sherrington Kirkpatrick (SK) model" [1].
+
+    The SK Hamiltonian over n spins is given as:
+    :math:`H(x)=-1/\sqrt{n} \sum_{i<j} w_{i,j}x_ix_j`,
+    where :math:`x_i\in\{\pm 1\}` and :math:`w_{i,j}\in\{\pm 1\}` are chosen independently and uniformly
+    at random. Notice that there are other variants e.g., with :math:`w_{i,j}` chosen from the normal
+    distribution with mean 0 and variance 1.
+
+    References:
+        [1]: Dmitry Panchenko. "The Sherrington-Kirkpatrick model: an overview",
+        https://arxiv.org/abs/1211.1094
+    """
+
+    def __init__(self, num_of_sites: int, rng: Optional[np.random.Generator] = None):
```

Comment:
One thing might make it easier for people if this were `Optional[Union[int, np.random.Generator]] = None` They can then simply pass an integer as a seed. They only need to go to a Generator if they need that level of control/definition over the random numbers. Then the code here to set up the self._rng can become something like
```
if isinstance(rng, np.random.Generator):
  self._rng = rng
else 
  self._rng = np.random.default_rng(rng)
```
The latter ie `np.random.default_rng(rng)` will get None or the int seed - where None ends up being seeded randomly from the OS by numpy.

Reply:
ok.

Reply:
I notice in the code you also did an extra check for `int` to separate it away from None. That's not really needed as `np.random.default_rng(None)` is equivalent to doing `np.random.default_rng()` which is why I allowed both int and None to flow through the else part above. In fact I chose to avoid doing an int check since there are various types e.g from numpy that can cause problems with something too specific.

Reply:
Thank you. Fixed.